File: libraryCoreJava/src/main/java/com/juqitech/moretickets/core/util/ToastUtil.java
package com.juqitech.moretickets.core.util;

import android.content.Context;
import android.view.Gravity;
import androidx.annotation.StringRes;
import android.widget.Toast;

/**
 * @author zhiqiang
 * @date 2019-06-04
 * @desc
 */
public class ToastUtil {

    private static ToastUtil td;

    public static void showLong(Context context, @StringRes int resId) {
        showLong(context, context.getString(resId));
    }

    public static void showLong(Context context, String msg) {
        if (td == null) {
            td = new ToastUtil(context);
        }
        td.setText(msg);
        td.create().show();
    }

    public static void showShort(Context context, @StringRes int resId) {
        showShort(context, context.getString(resId));
    }

    public static void showShort(Context context, String msg) {
        if (td == null) {
            td = new ToastUtil(context);
        }
        td.setText(msg);
        td.createShort().show();
    }

    private Context context;
    private Toast toast;
    private String msg;

    private ToastUtil(Context context) {
        this.context = context.getApplicationContext();
    }

    private Toast create() {
        if (null == toast) {
            toast = Toast.makeText(context, "", Toast.LENGTH_SHORT);

        }
        toast.setText(msg);
        toast.setGravity(Gravity.CENTER,0,0);
        toast.setDuration(Toast.LENGTH_LONG);
        return toast;
    }

    private Toast createShort() {
        if (null == toast) {
            toast = Toast.makeText(context, "", Toast.LENGTH_SHORT);
        }
        toast.setText(msg);
        toast.setGravity(Gravity.CENTER,0,0);
        toast.setDuration(Toast.LENGTH_SHORT);
        return toast;
    }

    private void setText(String text) {
        msg = text;
    }

}

File: libraryCore/src/main/java/com/juqitech/moretickets/core/util/ToastUtil.kt
package com.juqitech.moretickets.core.util

import android.content.Context
import android.view.Gravity
import androidx.annotation.StringRes
import android.widget.Toast

/**
 * @author zhanfeng
 * @date 2019-06-04
 * @desc
 */
class ToastUtil private constructor(context: Context) {

    private val context: Context = context.applicationContext
    private var toast: Toast? = null
    private var msg: String? = null

    private fun create(): Toast {
        if (null == toast) {
            toast = Toast.makeText(context, "", Toast.LENGTH_SHORT)

        }
        toast!!.setText(msg)
        toast!!.setGravity(Gravity.CENTER, 0, 0)
        toast!!.duration = Toast.LENGTH_LONG
        return toast!!
    }

    private fun createShort(): Toast {
        if (null == toast) {
            toast = Toast.makeText(context, "", Toast.LENGTH_SHORT)
        }
        toast!!.setText(msg)
        toast!!.setGravity(Gravity.CENTER, 0, 0)
        toast!!.duration = Toast.LENGTH_SHORT
        return toast!!
    }

    private fun setText(text: String) {
        msg = text
    }

    companion object {

        private var td: ToastUtil? = null

        @JvmStatic
        fun showLong(context: Context, @StringRes resId: Int) {
            showLong(context, context.getString(resId))
        }

        @JvmStatic
        fun showLong(context: Context, msg: String) {
            if (td == null) {
                td = ToastUtil(context)
            }
            td!!.setText(msg)
            td!!.create().show()
        }

        @JvmStatic
        fun showShort(context: Context, @StringRes resId: Int) {
            showShort(context, context.getString(resId))
        }

        @JvmStatic
        fun showShort(context: Context, msg: String) {
            if (td == null) {
                td = ToastUtil(context)
            }
            td!!.setText(msg)
            td!!.createShort().show()
        }
    }

}
